tests/pass/concurrency/scope.rs

```diff
@@ -0,0 +1,24 @@
+use std::thread;
+
+fn main() {
+    let mut a = vec![1, 2, 3];
+    let mut x = 0;
+
+    thread::scope(|s| {
+        s.spawn(|| {
+            // We can borrow `a` here.
+            let _s = format!("hello from the first scoped thread: {a:?}");
+        });
+        s.spawn(|| {
+            let _s = format!("hello from the second scoped thread");
+            // We can even mutably borrow `x` here,
+            // because no other threads are using it.
+            x += a[0] + a[2];
+        });
+        let _s = format!("hello from the main thread");
+    });
+
+    // After the scope, we can modify and access our variables again:
+    a.push(4);
+    assert_eq!(x, a.len());
+}
```

Comment:
Is this test safeguarding against anything in particular?

Reply:
No, I just realized we didn't have any test that calls scoped threads in our test suite here, and that seemed like something we should fix. And since scoped threads are their own sufficiently complicated thing, it felt better to make it a separate test (potentially to be extended with more ways of invoking scoped threads in the future).

I *think* this is a test that triggered https://github.com/rust-lang/miri/issues/2629 for me in the past, but I am not sure and anyway we don't have a good way of testing for https://github.com/rust-lang/miri/issues/2629.

Reply:
Not sure if I proposed this before... but we could use `./miri many-seeds` in CI? That's not proof against some strange scheduler situation, but it definitely hits a lot of code paths.

Reply:
We'd have to figure out which tests to run it on, though. 256 seeds takes quite a while so we can't do this for many tests.

Reply:
I would be happy with just this:
```rust
use std::thread;

fn main() {
  thread::scope(|s| {
    s.spawn(|| {});
  });
}
```

Reply:
I added that now. Even with just 64 seeds this takes a minute on my laptop... so it's a really expensive test. Not sure if we will keep it forever but I guess we can have it for now.

I counted 5 failing seeds below 64 before this PR so I hope this is enough coverage. (In the interval 64..128 there is only one failing test, so the 0..64 interval might just be lucky.)